packages/kbn-observable/src/observable.test.ts

```diff
@@ -0,0 +1,986 @@
+  });
+
+  test('can call "complete" on observer when no "complete" is defined when subscribed', () => {
+    let observer!: SubscriptionObserver<string>;
+    const cleanup = jest.fn();
+
+    const source = new Observable<string>(_observer => {
+      observer = _observer;
+      return cleanup;
+    });
+
+    source.subscribe();
+
+    expect(observer.complete()).toBe(undefined);
+    expect(cleanup).toHaveBeenCalledTimes(1);
+  });
+
+  test('does not call "complete" multiple times', () => {
+    let observer!: SubscriptionObserver<string>;
+
+    const source = new Observable<string>(_observer => {
+      observer = _observer;
+    });
+
+    const values: any[] = [];
+
+    source.subscribe({
+      complete() {
+        values.push('complete');
+      },
+    });
+
+    observer.complete();
+    observer.complete();
+
+    expect(values).toEqual(['complete']);
+  });
+
+  test('cleanup function is called when method throws', () => {
+    let observer!: SubscriptionObserver<string>;
+    const cleanup = jest.fn();
+
+    const consoleError = jest.spyOn(console, 'error');
+    consoleError.mockImplementation(noop);
+
+    const source = new Observable<string>(_observer => {
+      observer = _observer;
+      return cleanup;
+    });
+
+    source.subscribe({
+      complete() {
+        throw new Error('thrown from complete');
+      },
+    });
+
+    observer.complete();
+
+    expect(cleanup).toHaveBeenCalledTimes(1);
+    expect(consoleError.mock.calls).toMatchSnapshot('logs');
+
+    consoleError.mockRestore();
+  });
+
+  test('does not call "complete" on unsubscribe', () => {
+    let complete = jest.fn();
+
+    const source = new Observable(() => {});
+
+    source.subscribe({ complete }).unsubscribe();
+
+    expect(complete).not.toHaveBeenCalled();
+  });
+});
+
+describe('SubscriptionObserver#closed', () => {
+  test('returns false when the subscription is active', () => {
+    let isClosed;
+
+    new Observable(observer => {
+      isClosed = observer.closed;
+    }).subscribe();
+
+    expect(isClosed).toBe(false);
+  });
+
+  test('returns true when observer is completed', () => {
+    let isClosed;
+
+    new Observable(observer => {
+      observer.complete();
+
+      isClosed = observer.closed;
+    }).subscribe();
+
+    expect(isClosed).toBe(true);
+  });
+
+  test("returns true when observer has error'd", () => {
+    let isClosed;
+
+    new Observable(observer => {
+      observer.error(new Error('foo'));
+
+      isClosed = observer.closed;
+    }).subscribe();
+
+    expect(isClosed).toBe(true);
+  });
+
+  test('returns true after unsubscribe is called', () => {
+    let observer!: SubscriptionObserver<any>;
+
+    new Observable(_observer => {
+      observer = _observer;
+    })
+      .subscribe()
+      .unsubscribe();
+
+    expect(observer.closed).toBe(true);
+  });
+});
+
+describe('subscriptions', () => {
+  test('calls the cleanup function on unsubscribe', () => {
+    let callCount = 0;
```

Comment:
__nit:__ what about `jest.fn` instead of custom call counter (here and in tests below)?
```js
const mock = jest.fn();
const source = new Observable(() => mock);
...
expect(mock).toHaveBeenCalledTimes(1);
```